File: v2/src/main/kotlin/com/api/v2/payment/Payment.java
package com.api.v2.payment;

import com.api.v2.creditcard.CreditCard;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "v1_payments")
public record Payment(
        CreditCard creditCard,
        String paidAt,
        Rent rent
) {
}

File: v2/src/main/kotlin/com/api/v2/payment/Payment.kt
package com.api.v2.payment

import com.api.v2.car.domain.Car
import com.api.v2.creditcard.domain.CreditCard
import com.api.v2.customer.domain.Customer
import org.springframework.data.mongodb.core.mapping.Document
import java.time.ZonedDateTime

@Document(collection = "v2_payments")
data class Payment(
    val customer: Customer,
    val car: Car,
    val creditCard: CreditCard,
    val paidAt: String
) {

    constructor(
        customer: Customer,
        car: Car,
        creditCard: CreditCard
    ): this (
        customer,
        car,
        creditCard,
        ZonedDateTime.now().toString()
    )

}
